applications/ConstitutiveLawsApplication/custom_constitutive/associative_plastic_damage_model.cpp

```diff
@@ -627,11 +651,19 @@ void AssociativePlasticDamageModel<TYieldSurfaceType>::IntegrateStressPlasticDam
 
                 // Compute the compliance increment -> C dot
                 CalculateComplianceMatrixIncrement(rValues, rPDParameters);
-                noalias(rPDParameters.ComplianceMatrix) += rPDParameters.ComplianceMatrixIncrement;
 
                 noalias(rPDParameters.StressVector) -= rPDParameters.PlasticConsistencyIncrement * prod(rPDParameters.ConstitutiveMatrix, rPDParameters.PlasticFlow);
 
-                CalculateConstitutiveMatrix(rValues, rPDParameters);
+                if (crack_reclosure) {
+                    // now we check the tensile-compressive distribution
+                    GenericConstitutiveLawIntegratorPlasticity<TYieldSurfaceType>::CalculateIndicatorsFactors(rPDParameters.StressVector, tension_parameter,compression_parameter);
+                    noalias(rPDParameters.ComplianceMatrix) += tension_parameter*rPDParameters.ComplianceMatrixIncrement;
+                    noalias(rPDParameters.ComplianceMatrixCompression) += compression_parameter*rPDParameters.ComplianceMatrixIncrement;
+                    MathUtils<double>::InvertMatrix(tension_parameter*rPDParameters.ComplianceMatrix+compression_parameter*rPDParameters.ComplianceMatrixCompression, rPDParameters.ConstitutiveMatrix, det);
```

Comment:
Again

Reply:
I can create a local variable :) @RiccardoRossi do you think that it would be more efficient?